http-server/src/main/java/org/triplea/server/http/web/socket/SessionSet.java

```diff
@@ -0,0 +1,76 @@
+package org.triplea.server.http.web.socket;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.Collection;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import javax.websocket.Session;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.triplea.server.lobby.chat.InetExtractor;
+
+/**
+ * Tracks sessions, can be used to listen to IP session ban events and close any matching sessions.
+ *
+ * <p>Closed sessions are removed when sessions are retrieved, so there is no need to 'remove' or
+ * 'unregister' sessions.
+ *
+ * <ol>
+ *   This class has two main purposes:
+ *   <li>Unify common session tracking code
+ *   <li>Provide a mechanism to close already established connections that have been banned. We will
+ *       rely on request filtering to prevent banned sessions from creating a new websocket
+ *       connection, this part is needed to remove existing connections of a banned user.
+ * </ol>
+ */
+@Slf4j
+public class SessionSet {
```

Comment:
You could add a `@ThreadSafe` annotation to this class as some sort of documentation